### PR TITLE
Users api

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Trello Card Link
https://trello.com/c/A7BOsv5J/1476-port-the-new-api-into-the-dapp

### Description
Moves profile pages, artist tracks on profiles & artist tracks on track pages over to new API endpoints via API client.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
Touches profile pages

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

Manually pointed discprov at prod dataset & ran dapp against. Visited profile pages & tracks page.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-users-api

File: src/containers/track-page/store/sagas.js [hunk 1]
```diff
@@ -144,6 +142,7 @@ function* watchFetchTrackSucceeded() {
 
 function* watchRefetchLineup() {
   yield takeEvery(trackPageActions.REFETCH_LINEUP, function* (action) {
+    console.log('refetch')
```
Comment (raymondjacobson) [inline src/containers/track-page/store/sagas.js:145]: removing

File: src/store/cache/users/sagas.js [hunk 2]
```diff
@@ -39,6 +42,19 @@ export function* fetchUsers(
   })
 }
 
+function* retrieveUserByHandle(handle) {
+  const userId = yield select(getUserId)
+  if (Array.isArray(handle)) {
+    handle = handle[0]
+  }
+  const encodedUserId = userId ? encodeHashId(userId) ?? undefined : undefined
```
Comment (jowlee) [inline src/store/cache/users/sagas.js:50]: I think we moved this conversion to the apiClient

File: src/containers/track-page/store/sagas.js (hunk 1 above)
Comment (jowlee) [inline src/containers/track-page/store/sagas.js]: log

File: src/containers/profile-page/store/lineups/tracks/retrieveUserTracks.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export function* retrieveUserTracks({
+  handle,
+  currentUserId,
+  sort,
+  offset,
+  limit
+}: RetrieveUserTracksArgs): Generator<any, Track[], any> {
+  const encodedUserId = currentUserId
+    ? encodeHashId(currentUserId) ?? undefined
```
Comment (jowlee) [inline src/containers/profile-page/store/lineups/tracks/retrieveUserTracks.ts:23]: moving encoding to be contained in apiclient

**Review by jowlee — APPROVED** [review]

File: src/store/cache/users/sagas.js (hunk 2 above)
Comment (raymondjacobson) [reply src/store/cache/users/sagas.js:50]: thx good catch

**Pull request merged**